server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -483,192 +541,85 @@ describe("Routerlicious", () => {
-
-									clients.forEach((client, index) => {
-										assert.equal(
-											client.signalCount,
-											1,
-											`User ${index + 1} should have received 1 signal`,
-										);
-										assert.equal(
-											userSignals[index][0].content,
-											numberSignalContent,
-											`User ${index + 1} signal content mismatch`,
-										);
-									});
-								});
-								it("should broadcast signal with boolean content", async () => {
-									const booleanSignalContent = true;
-									clients[0].socket.send("submitSignal", clients[0].clientId, [
-										booleanSignalContent,
-									]);
-									const userSignals = await Promise.all(clientSignalPromises);
-
-									clients.forEach((client, index) => {
-										assert.equal(
-											client.signalCount,
-											1,
-											`User ${index + 1} should have received 1 signal`,
-										);
-										assert.equal(
-											userSignals[index][0].content,
-											booleanSignalContent,
-											`User ${index + 1} signal content mismatch`,
-										);
-									});
-								});
-								it("should broadcast signal with JSON/Object content", async () => {
-									const objectSignalContent = {
-										key1: "value1",
-										key2: 42,
-										key3: true,
-									};
-									clients[0].socket.send("submitSignal", clients[0].clientId, [
-										objectSignalContent,
+							[null, "invalid"].forEach((clientId) => {
+								it(`should nack signals with ${clientId} client ID`, () => {
+									listenForNacks(clients[0]);
+									clients[0].socket.send("submitSignal", clientId, [
+										stringSignalContent,
 									]);
-									const userSignals = await Promise.all(clientSignalPromises);
-
-									clients.forEach((client, index) => {
-										assert.equal(
-											client.signalCount,
-											1,
-											`User ${index + 1} should have received 1 signal`,
-										);
-										assert.deepEqual(
-											userSignals[index][0].content,
-											objectSignalContent,
-											`User ${index + 1} signal content mismatch`,
-										);
-									});
+									checkNack(clients[0], "Nonexistent client");
 								});
 							});
-						});
 
-						describe("sending two signals", () => {
-							beforeEach(() => {
-								clientSignalPromises = clients.map((client) =>
-									createSignalPromise(client, 2),
+							it("nacks signal that is not an array", () => {
+								listenForNacks(clients[0]);
+
+								clients[0].socket.send(
+									"submitSignal",
+									clients[0].clientId,
+									stringSignalContent,
 								);
+								checkNack(clients[0], "Invalid signal message");
 							});
 
-							it("should broadcast signals sent from multiple clients to all connected clients", async () => {
-								clients[0].socket.send("submitSignal", clients[0].clientId, [
-									"first signal",
-								]);
+							[
+								42,
+								true,
```

Comment:
Let's add a string case to have local completeness.

Reply:
Added